src/opdcodes.rs

```diff
@@ -308,6 +424,48 @@ pub fn not(vm: &mut VM, instruction: u16) -> Result<(), VMError> {
     Ok(())
 }
 
+pub fn store(vm: &mut VM, instruction: u16) -> Result<(), VMError> {
+    let sr = (instruction >> 9) & 0x7;
+    let pc_offset = sign_extend(instruction & 0x1FF, 9);
+
+    let address = vm.registers.pc.wrapping_add(pc_offset);
+
+    let value = vm.registers.get(sr.into())?;
+
+    vm.write_memory(address, value)?;
+
+    Ok(())
+}
+
+pub fn store_indirect(vm: &mut VM, instruction: u16) -> Result<(), VMError> {
+    let sr = (instruction >> 9) & 0x7;
+    let pc_offset = sign_extend(instruction & 0x1FF, 9);
```

Comment:
Btw, just out of curiosity, why are you using hexadecimal numbers for bit masking?

Maybe using a binary number would be a bit clearer, something like `instruction & 0b000_0001_1111_1111`.

It's just a suggestion! If you prefer hexadecimals keep them, I just think it'd be easier to read if you could see the bit mask as bits.

Reply:
I see your point, plus copying and pasting the flags assures that you don't make any mistakes while writing the number down.

As you wish!